src/libraries/System.Private.CoreLib/src/System/IO/RandomAccess.cs

```diff
@@ -250,7 +251,44 @@ public static ValueTask WriteAsync(SafeFileHandle handle, IReadOnlyList<ReadOnly
             return WriteGatherAtOffsetAsync(handle, buffers, fileOffset, cancellationToken);
         }
 
-        private static void ValidateInput(SafeFileHandle handle, long fileOffset)
+        /// <summary>
+        /// Flushes the operating system buffers for the given file to disk.
+        /// </summary>
+        /// <param name="handle">The file handle.</param>
+        /// <exception cref="T:System.ArgumentNullException"><paramref name="handle" /> is <see langword="null" />.</exception>
+        /// <exception cref="T:System.ArgumentException"><paramref name="handle" /> is invalid.</exception>
+        /// <exception cref="T:System.ObjectDisposedException">The file is closed.</exception>
+        /// <exception cref="T:System.UnauthorizedAccessException">
+        /// <paramref name="handle" /> was not opened for writing (on Windows).
+        /// </exception>
+        /// <exception cref="T:System.IO.IOException">An I/O error occurred.</exception>
+        /// <remarks>
+        /// <para>
+        /// This method calls platform-dependent APIs such as <c>FlushFileBuffers()</c> on Windows and <c>fsync()</c> on Unix.
+        /// </para>
+        /// <para>
+        /// Flushing the buffers causes data to be written to disk which is a relatively expensive operation. It is recommended
```

Comment:
Thanks @adamsitnik! I was a bit nervous about writing such detailed comments (since the default for much of the libraries is to have no comments at all) but I am glad some of the comments have proven useful 🙏